### ISSUE TITLE
Discrepancy between documentation and implementation

### ISSUE THREAD

**Issue by rbonichon** [mirage/ocaml-hex#36]
The documentation  of `of_string` states
```
val of_string: ?ignore:char list -> string -> t
(** [of_string s] is the hexadecimal representation of the binary
    string [s]. If [ignore] is set, skip the characters in the list
    when converting. Eg [of_string ~ignore:[' '] "a f"]. The default
    value of [ignore] is [[]]). *)
```

while the implemenation
```
let of_string ?ignore s =
  match ignore with
    None -> of_string_fast s
  | Some ignore -> of_helper ~ignore (fun i -> s.[i]) (String.length s)
```

Clearly the default value of ignore is not `[]` - there's none. While it may behave as if it were (I expect `of_helper []` to be functionally equivalent to `of_string_fast` -- but haven't checked). 

I see two choices here:
1. set up the default value as stated in the documentation and pattern-match on it - thus having `of_string_fast` handling the `[]` case
2. Rewrite the documentation to match the implementation.

I don't know which you'd favor (I think I'd prefer 1) but let me know and I'll do a PR

**Comment by rbonichon** [comment]
Probably `of_bytes` needs to be changed as well to reflect the choice

**Comment by hannesm** [comment]
thanks for your report, fixed in [bcc5399](https://github.com/hannesm/ocaml-hex/commit/bcc5399f0b51f7306cf26e0e5727c38906224de2)
